src/mlio/csv_reader.cxx

```diff
@@ -289,26 +289,33 @@ csv_reader::init_parsers_and_schema()
     auto col_beg = tbb::make_zip_iterator(idx_beg, name_beg, type_beg);
     auto col_end = tbb::make_zip_iterator(idx_end, name_end, type_end);
 
+    std::unordered_map<std::string, std::size_t> name_count{};
+
     for (auto col_pos = col_beg; col_pos < col_end; ++col_pos) {
         std::size_t idx = std::get<0>(*col_pos);
-
-        std::string const &name = std::get<1>(*col_pos);
-
+        std::string name = std::get<1>(*col_pos);
         data_type dt = std::get<2>(*col_pos);
 
         if (should_skip(idx, name)) {
             skipped_columns_.emplace_back(1);
-
             column_parsers_.emplace_back(nullptr);
+            continue;
         }
-        else {
-            skipped_columns_.emplace_back(0);
 
-            column_parsers_.emplace_back(make_parser(dt, params_.parser_prm));
+        skipped_columns_.emplace_back(0);
+        column_parsers_.emplace_back(make_parser(dt, params_.parser_prm));
 
-            descs.emplace_back(
-                feature_desc_builder{name, dt, {batch_size, 1}}.build());
+        if (params_.dedupe_column_names) {
```

Comment:
Could you please briefly describe the logic here.

Reply:
I added a brief comment.